_posts/en/newsletters/2020-04-29-newsletter.md

```diff
@@ -0,0 +1,265 @@
+  commitment transactions to be [CPFP][topic cpfp] fee bumped via
+  [anchor outputs][topic anchor outputs].  We'll describe the attack
+  using an extended example: Alice uses an LN channel to send Bob a Hash
+  Time Lock Contract (HTLC) which is designed to be settled in either of
+  the following ways:
+
+     - If Bob discloses the preimage for `<hash>`, he can spend 1 BTC of
+       Alice's money
+
+     - Otherwise, after 80 blocks, Alice can spend that 1 BTC back to
+       herself
+
+    Alice also told Bob that the goal of her payment is to pay Mallory, so Bob
+    uses a channel he has with Mallory to send her a related HTLC:
+
+     - If Mallory discloses the preimage for `<hash>`, she can spend 1
+       BTC of Bob's money (we're ignoring routing fees in this example)
+
+     - Otherwise, after 40 blocks, Bob can spend his 1 BTC back to
+       himself
+
+    Although the above HTLCs are usually created and settled offchain,
+    each party also has a *commitment transaction* they can use to
+    put the HTLC commitment onchain.  A separate onchain *settlement
+    transaction* can fulfill either condition of the HTLC.
+
+    For example, Mallory can publish the commitment transaction and then
+    create a settlement transaction that provides the preimage and
+    claims Bob's 1 BTC.  If Bob sees Mallory's preimage settlement
+    transaction before the 80 block timeout from the Alice-Bob contract,
+    Bob can extract that preimage and use it to claim the 1 BTC from
+    Alice (either onchain or offchain).  Or, if Bob doesn't see a
+    preimage settlement transaction, Bob can create his own refund
+    settlement transaction after 40 blocks that takes back his 1 BTC,
+    allowing him to also refund Alice her 1 BTC (again, either onchain
+    or offchain).  In either case, this leaves everyone in compliance
+    with the intent of their contracts.
+
+    Unfortunately, as disclosed this week, there appears to be a way for
+    Mallory to circumvent the process by both preventing Bob from
+    learning the preimage while also preventing him from sending his
+    refund settlement transaction.
+
+    - **Preimage denial:** Mallory can prevent Bob from learning the
+      preimage by giving her preimage settlement transaction a low
+      feerate that keeps it from being confirmed quickly.  If Bob is
+      only looking for preimages in the block chain, he won't see
+      Mallory's transaction while it remains unconfirmed.
+
+    - **Refund denial:** Mallory's prior broadcast of the preimage
+      settlement transaction can prevent miners and Bitcoin relay nodes
+      from accepting Bob's later broadcast of the refund settlement
+      transaction because the two transactions *conflict*, meaning they
+      both spend the same input (a UTXO created in the commitment
+      transaction).  In theory, Bob's refund settlement transaction will
+      pay a higher feerate and so can [replace][topic rbf] Mallory's
```

Comment:
At least *currently*, if *Bob* drops its commitment transaction onchain, then Bob cannot RBF its timelocked refund settlement transaction (as it requires a signature from Mallory, which was negotiated before while setting up the HTLC).  Mallory can arrange this to happen.

However this is arguably a minor detail: even *with* "full" RBF, due to the non-consensual state of the mempool, various pinning attacks can still be done, most notably by setting up a large chain of transactions with horrid feerate but high absolute fee, which cannot be evicted except with an even higher high absolute fee on the replacing transaction, and racing this with the victim Bob, such that miners get this large chain, but non-mining nodes cannot.

This might be more detail than you care to add though.